testing/go/framework.go

```diff
@@ -154,6 +156,15 @@ func runScript(t *testing.T, ctx context.Context, script ScriptTest, conn *pgx.C
 				require.NoError(t, err)
 				readRows, err := ReadRows(rows, normalizeRows)
 				require.NoError(t, err)
+
+				if assertion.Cols != nil {
+					fields := rows.FieldDescriptions()
+					require.Len(t, fields, len(assertion.Cols))
```

Comment:
`if assert.Len(...) {...}` instead of the `require` here. We should still check for the rows below if the column field count fails.